bitwarden_license/src/Commercial.Infrastructure.EntityFramework/Repositories/SecretRepository.cs

```diff
@@ -44,9 +44,20 @@ public async Task<IEnumerable<Core.Entities.Secret>> GetManyByOrganizationIdAsyn
             var dbContext = GetDatabaseContext(scope);
             var secrets = await dbContext.Secret
                                     .Where(c => c.OrganizationId == organizationId && c.DeletedDate == null)
+                                    .Include("Projects")
                                     .OrderBy(c => c.RevisionDate)
                                     .ToListAsync();
-            return Mapper.Map<List<Core.Entities.Secret>>(secrets);
+
+            var secretsWithProjectDetails = new List<Core.Entities.Secret>();
+
+            foreach (var secret in secrets)
+            {
+                secret.ProjectGuids = secret.Projects?.ToDictionary(x => x.Id, x => x.Name);
+                var secretWithProjectDetails = Mapper.Map<Core.Entities.Secret>(secret);
+                secretsWithProjectDetails.Add(secretWithProjectDetails);
+            }
```

Comment:
I'm wondering if we need to do this logic in the repository layer, or if we should do it in the response models.

Reply:
I started making these changes to see what this would look like, It was getting messy and I stashed the changes - if you want to continue on this route we should go over my stash and see if it's what you imagined.